codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2StreamFrame.java

```diff
@@ -50,11 +45,15 @@ public boolean equals(Object o) {
             return false;
         }
         Http2StreamFrame other = (Http2StreamFrame) o;
-        return streamId == other.streamId();
+        return stream == other.stream() || (stream != null && stream.equals(other.stream()));
     }
 
     @Override
     public int hashCode() {
```

Comment:
From https://github.com/netty/netty/pull/6896#discussion_r132118328

@normanmaurer 
> yeah only for testing. Let me think about what to do with it. Maybe we should just remove it all together and act on the default implementation. WDYT ?

I think it is fine and useful for testing ... but we should consider clarifying via javadocs the intention of these methods (designed to support testing and not necessarily indented for general purpose use ... use at your own risk).